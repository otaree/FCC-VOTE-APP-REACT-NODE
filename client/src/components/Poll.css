.Poll_body {
    width: 80%;
    margin: 2% auto;
}

.Poll_header h2, .Poll_header p {
    text-align: center;
}

.Poll_header .Poll_control {
    text-align: center;
}

.Poll_header .Poll_control a {
    text-decoration: none;
    border: thin solid #bbb;
    border-radius: 3px;
    padding: 5px 10px;
    color: #eee;
    background-color: rgb(199, 71, 71);
    font-size: 16px;
    font-weight: bold;
}