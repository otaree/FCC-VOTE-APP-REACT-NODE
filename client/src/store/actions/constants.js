// auth
export const AUTH_LOADING = "AUTH_LOADING";
export const AUTH_GET_UID = "AUTH_GET_UID";
export const AUTH_LOGIN = "AUTH_LOGIN";
export const AUTH_LOGOUT = "AUTH_LOGOUT";
export const AUTH_FAIL = "AUTH_FAIL";

// poll
export const POLL_LOADING = "POLL_LOADING";
export const POLL_SUCCESS = "POLL_SUCCESS";
export const POLL_FAIL = "POLL_FAIL";
export const POLL_SET_POLLS = "POLL_SET_POLLS";
export const POLL_SET = "POLL_SET";
export const POLL_UNSET = "POLL_UNSET";