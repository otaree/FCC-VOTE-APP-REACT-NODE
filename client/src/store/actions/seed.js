export const polls = [
    {
        _id: "1",
        question: "Yes or No?",
        createdAt: 1000,
        author: '123abc',
        voters: ["123abc"],
        options: [{
            votes: 1,
            value: 'Yes',
            _id: "101"
        }, {
            votes: 0,
            value: 'No',
            _id: "102"
        }]
    },
    {
        _id: "2",
        question: "2017 or 2018?",
        createdAt: 2000,
        author: '124abc',
        voters: ["124abc"],
        options: [{
            votes: 1,
            value: '2018',
            _id: "201"
        }, {
            votes: 0,
            value: '2017',
            _id: "202"
        }]
    }
];