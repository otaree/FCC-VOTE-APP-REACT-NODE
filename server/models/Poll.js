const mongoose = require('mongoose');

const PollSchema = new mongoose.Schema({
    question: {
        type: String,
        required: true,
        minlength: 1,
        unique: true,
        trim: true
    },
    createdAt: {
        type: Number,
        default: new Date().getTime()
    },
    author: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User'
    },
    voters: [
        {
            voter: {
                type: mongoose.Schema.Types.ObjectId,
                ref: 'User',
            }
        }
    ],
    options: [
        {
            votes: {
                type: Number,
                default: 0
            },
            value: {
                type: String,
                required: true,
                minlength: 1,
                trim: true
            }
        }
    ]
});

// PollSchema.statics.casteVote = function (poll_id, option_id, user_id) {
//     const Poll = this;

//     return new Promise((resolve, reject) => {
//         Poll.findOne({ _id: poll_id, "options._id": option_id })
//             .then(poll => {
//                 if (poll) {
//                     return reject("User already voted");
//                 }

//             })
//     });
// } 

const Poll = mongoose.model('Poll', PollSchema);

module.exports = { Poll };