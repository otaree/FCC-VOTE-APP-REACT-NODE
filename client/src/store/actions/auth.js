import axios from 'axios';
import uniqid from 'uniqid';

import * as constants from './constants';

export const loading = () => {
    return {
        type: constants.AUTH_LOADING,
        loading: true
    };
};

export const login = (user) => {
    return {
        ...user,
        loading: false,
        error: {
            isError: false,
            value: ''
        },
        type: constants.AUTH_LOGIN
    }
};

export const logout = () => {
    return {
        type: constants.AUTH_LOGOUT,
        loading: false,
        error: {
            isError: false,
            value: ''
        }
    };
};

export const fail = (value) => {
    return {
        type: constants.AUTH_FAIL,
        loading: false,
        error: {
            isError: true,
            value
        }
    };
};

export const initUser = (token) => {
    return async dispatch => {
        dispatch(loading());
        try {
            const response = await axios({ url: 'http://localhost:5000/user/token', method: 'get', headers: { 'x-auth': token }});
            dispatch(login(response.data));
        } catch (e) {
            dispatch(fail("Authentication fail"));
        }
    };
};

export const getUid = () => {
    let uid = localStorage.getItem("uid");
    if (!uid) {
        uid = uniqid();
        localStorage.setItem("uid", uid)
    }
    return {
        type: constants.AUTH_GET_UID,
        uid
    }
};