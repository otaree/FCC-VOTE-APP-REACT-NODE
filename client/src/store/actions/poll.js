import axios from 'axios';

import * as constants from './constants';

export const pollLoading = () => ({
    type: constants.POLL_LOADING,
    loading: true
});

export const pollSuccess = () => ({
    type: constants.POLL_SUCCESS,
    loading: false,
    error: {
        isError: false,
        value: ''
    }
});

export const pollFail = value => ({
    type: constants.POLL_FAIL,
    loading: false,
    error: {
        isError: true,
        value
    }
});

export const setPolls = polls => ({
    type: constants.POLL_SET_POLLS,
    loading: false,
    error: {
        isError: false,
        value: ''
    },
    polls
});

export const setPoll = poll => ({
    type: constants.POLL_SET,
    loading: false,
    error: {
        isError: false,
        value: ''
    },
    poll
});

export const fetchPolls = () => {
    return async dispatch => {
        dispatch(pollLoading());

        try {
            const response = await axios.get('http://localhost:5000/polls');
            dispatch(setPolls(response.data));
            return Promise.resolve("PASS");
        } catch (e) {
            dispatch(pollFail("Network Problem"));
            return Promise.reject("FAIL");
        }
    }
};

export const fetchPoll = (id) => {
    return async dispatch => {
        dispatch(pollLoading());

        try {
            const response = await axios.get(`http://localhost:5000/poll/${id}`);
            dispatch(setPoll(response.data));
            return Promise.resolve();
        } catch (e) {
            dispatch(pollFail("Network Problem"));
            return Promise.reject();
        }
    };
};

export const createPoll = (poll, token) => {
    return async dispatch => {
        dispatch(pollLoading());

        try {
            const response = await axios({ url: "http://localhost:5000/poll", method: "post", headers: { "x-auth": token }, data: poll });
            dispatch(pollSuccess());
            return response.data;
        } catch (e) {
            dispatch(pollFail("Network Problem"));
            return Promise.reject();
        }
    }
};

export const updatePoll = (poll, token, id) => {
    return async dispatch => {
        dispatch(pollLoading());

        try {
            const response = await axios({ url: `http://localhost:5000/poll/${id}`, method: "patch", headers: { "x-auth": token }, data: poll });
            dispatch(pollSuccess());
            return response.data;
        } catch (e) {
            dispatch(pollFail("Network Problem"));
            return Promise.reject();
        }
    }
};

export const deletePoll = (token, id) => {
    return async dispatch => {
        dispatch(pollLoading());

        try {
            const response = await axios({ url: `http://localhost:5000/poll/${id}`, method: "delete", headers: { "x-auth": token } });
            dispatch(pollSuccess());
            return response.data;
        } catch (e) {
            dispatch(pollFail("Network Problem"));
            return Promise.reject();
        }
    }
};

export const votePoll = (id, optionId, userId) => {
    return async dispatch => {
        dispatch(pollLoading());

        try {
            const response = await axios({ url: `http://localhost:5000/poll/${id}/vote`, method: "patch", data: { option: optionId, userId } });
            dispatch(pollSuccess());
            return response.data;
        } catch (e) {
            dispatch(pollFail("Network Problem"));
            return Promise.reject();
        }
    }
};