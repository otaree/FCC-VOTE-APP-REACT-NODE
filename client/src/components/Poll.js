import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { isEmpty } from 'lodash';

import Chart from './Chart';
import VotePoll from './VotePoll';
import * as pollActions from '../store/actions/poll';

export class Poll extends React.Component {
    async componentDidMount() {
        const id = this.props.match.params.id;
        await this.props.fetchPoll(id);
    }

    componentWillUnmount() {
        this.props.unsetPoll();
    }

    voteHandler = async (optionsId) => {
        try {
            const id = this.props.match.params.id;
            const poll = await this.props.vote(id, optionsId, this.props.uid);
            this.props.setPoll(poll);
        } catch (e) {

        }
    };

    render() {
        let poll = <h2 style={{ textAlign: "center"}}>Loading...</h2>;
        let editButton;
        if (this.props.poll) {
            if (this.props.userId) {
                editButton = this.props.userId === this.props.poll.author._id ? (
                    <div className="level-item ">
                        <Link className="button is-info" to={`/poll/${this.props.poll._id}/edit`}>Edit</Link>
                    </div>
                        ) : null;
            }
            poll = (
                <section className="section">
                    <div className="container">
                        <div className="box">
                            <div className="level">
                                <div className="level-item">
                                    <h2 className="title">{this.props.poll.question}</h2>                                        
                                </div>
                            </div>
                                <div className="level">
                                    <div className="level-item">
                                        <p className="subtitle">by {this.props.poll.author.name}</p>                                            
                                    </div>
                                </div>
                            <div className="level">
                                <div className="level-item">
                                    {editButton}                                        
                                </div>
                            </div>
                            {
                                this.props.poll.voters.indexOf(this.props.uid) === -1 ? (
                                    <VotePoll poll={this.props.poll} onVote={this.voteHandler}/>
                                ): (
                                    <Chart poll={this.props.poll} />
                                )
                            }
                        </div>
                    </div>
                </section>
            );
        }
        return poll;
    }
}

const mapStateToProps = state => {
    return {
        loading: state.poll.loading,
        poll: isEmpty(state.poll.poll) ? null : state.poll.poll,
        uid: state.auth.uid,
        userId: state.auth.userId
    };
}

const mapDispatchToProps = dispatch => {
    return {
        fetchPoll: (id) => dispatch(pollActions.fetchPoll(id)),
        vote: (id, optionId, uid) => dispatch(pollActions.votePoll(id, optionId, uid)),
        setPoll: (poll) => dispatch(pollActions.setPoll(poll)),
        unsetPoll: () => dispatch(pollActions.unsetPoll())
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(Poll);