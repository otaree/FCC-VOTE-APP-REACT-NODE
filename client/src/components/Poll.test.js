import React from 'react';
import { shallow } from 'enzyme';

import { Poll } from './Poll';
import { polls } from '../store/actions/seed';

describe("Poll", () => {
    const props = {
        poll: polls[0]
    };
    const poll = shallow(<Poll {...props} />);

    it('renders correctly', () => {
        expect(poll).toMatchSnapshot();
    });
});