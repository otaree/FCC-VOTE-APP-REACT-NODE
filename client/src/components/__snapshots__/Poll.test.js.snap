// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Poll renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Poll
    poll={
        Object {
            "_id": "1",
            "author": "123abc",
            "createdAt": 1000,
            "options": Array [
              Object {
                "_id": "101",
                "value": "Yes",
                "votes": 1,
              },
              Object {
                "_id": "102",
                "value": "No",
                "votes": 0,
              },
            ],
            "question": "Yes or No?",
            "voters": Array [
              "123abc",
            ],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container"
>
        <div
                className="box"
        >
                <div
                        className="level"
                >
                        <div
                                className="level-item"
                        >
                                <h2
                                        className="title"
                                >
                                        Yes or No?
                                </h2>
                        </div>
                </div>
                <div
                        className="level"
                >
                        <div
                                className="level-item"
                        >
                                <p
                                        className="subtitle"
                                >
                                        by 
                                </p>
                        </div>
                </div>
                <div
                        className="level"
                >
                        <div
                                className="level-item"
                        />
                </div>
                <VotePoll
                        onVote={[Function]}
                        poll={
                                Object {
                                        "_id": "1",
                                        "author": "123abc",
                                        "createdAt": 1000,
                                        "options": Array [
                                          Object {
                                            "_id": "101",
                                            "value": "Yes",
                                            "votes": 1,
                                          },
                                          Object {
                                            "_id": "102",
                                            "value": "No",
                                            "votes": 0,
                                          },
                                        ],
                                        "question": "Yes or No?",
                                        "voters": Array [
                                          "123abc",
                                        ],
                                      }
                        }
                />
        </div>
</div>,
      "className": "section",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="box"
>
          <div
                    className="level"
          >
                    <div
                              className="level-item"
                    >
                              <h2
                                        className="title"
                              >
                                        Yes or No?
                              </h2>
                    </div>
          </div>
          <div
                    className="level"
          >
                    <div
                              className="level-item"
                    >
                              <p
                                        className="subtitle"
                              >
                                        by 
                              </p>
                    </div>
          </div>
          <div
                    className="level"
          >
                    <div
                              className="level-item"
                    />
          </div>
          <VotePoll
                    onVote={[Function]}
                    poll={
                              Object {
                                        "_id": "1",
                                        "author": "123abc",
                                        "createdAt": 1000,
                                        "options": Array [
                                          Object {
                                            "_id": "101",
                                            "value": "Yes",
                                            "votes": 1,
                                          },
                                          Object {
                                            "_id": "102",
                                            "value": "No",
                                            "votes": 0,
                                          },
                                        ],
                                        "question": "Yes or No?",
                                        "voters": Array [
                                          "123abc",
                                        ],
                                      }
                    }
          />
</div>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="level"
>
              <div
                            className="level-item"
              >
                            <h2
                                          className="title"
                            >
                                          Yes or No?
                            </h2>
              </div>
</div>,
            <div
              className="level"
>
              <div
                            className="level-item"
              >
                            <p
                                          className="subtitle"
                            >
                                          by 
                            </p>
              </div>
</div>,
            <div
              className="level"
>
              <div
                            className="level-item"
              />
</div>,
            <VotePoll
              onVote={[Function]}
              poll={
                            Object {
                                          "_id": "1",
                                          "author": "123abc",
                                          "createdAt": 1000,
                                          "options": Array [
                                            Object {
                                              "_id": "101",
                                              "value": "Yes",
                                              "votes": 1,
                                            },
                                            Object {
                                              "_id": "102",
                                              "value": "No",
                                              "votes": 0,
                                            },
                                          ],
                                          "question": "Yes or No?",
                                          "voters": Array [
                                            "123abc",
                                          ],
                                        }
              }
/>,
          ],
          "className": "box",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="level-item"
>
                <h2
                                className="title"
                >
                                Yes or No?
                </h2>
</div>,
              "className": "level",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h2
                  className="title"
>
                  Yes or No?
</h2>,
                "className": "level-item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Yes or No?",
                  "className": "title",
                },
                "ref": null,
                "rendered": "Yes or No?",
                "type": "h2",
              },
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="level-item"
>
                <p
                                className="subtitle"
                >
                                by 
                </p>
</div>,
              "className": "level",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <p
                  className="subtitle"
>
                  by 
</p>,
                "className": "level-item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "by ",
                    undefined,
                  ],
                  "className": "subtitle",
                },
                "ref": null,
                "rendered": Array [
                  "by ",
                  undefined,
                ],
                "type": "p",
              },
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="level-item"
/>,
              "className": "level",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "level-item",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onVote": [Function],
              "poll": Object {
                "_id": "1",
                "author": "123abc",
                "createdAt": 1000,
                "options": Array [
                  Object {
                    "_id": "101",
                    "value": "Yes",
                    "votes": 1,
                  },
                  Object {
                    "_id": "102",
                    "value": "No",
                    "votes": 0,
                  },
                ],
                "question": "Yes or No?",
                "voters": Array [
                  "123abc",
                ],
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
>
          <div
                    className="box"
          >
                    <div
                              className="level"
                    >
                              <div
                                        className="level-item"
                              >
                                        <h2
                                                  className="title"
                                        >
                                                  Yes or No?
                                        </h2>
                              </div>
                    </div>
                    <div
                              className="level"
                    >
                              <div
                                        className="level-item"
                              >
                                        <p
                                                  className="subtitle"
                                        >
                                                  by 
                                        </p>
                              </div>
                    </div>
                    <div
                              className="level"
                    >
                              <div
                                        className="level-item"
                              />
                    </div>
                    <VotePoll
                              onVote={[Function]}
                              poll={
                                        Object {
                                                  "_id": "1",
                                                  "author": "123abc",
                                                  "createdAt": 1000,
                                                  "options": Array [
                                                    Object {
                                                      "_id": "101",
                                                      "value": "Yes",
                                                      "votes": 1,
                                                    },
                                                    Object {
                                                      "_id": "102",
                                                      "value": "No",
                                                      "votes": 0,
                                                    },
                                                  ],
                                                  "question": "Yes or No?",
                                                  "voters": Array [
                                                    "123abc",
                                                  ],
                                                }
                              }
                    />
          </div>
</div>,
        "className": "section",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="box"
>
            <div
                        className="level"
            >
                        <div
                                    className="level-item"
                        >
                                    <h2
                                                className="title"
                                    >
                                                Yes or No?
                                    </h2>
                        </div>
            </div>
            <div
                        className="level"
            >
                        <div
                                    className="level-item"
                        >
                                    <p
                                                className="subtitle"
                                    >
                                                by 
                                    </p>
                        </div>
            </div>
            <div
                        className="level"
            >
                        <div
                                    className="level-item"
                        />
            </div>
            <VotePoll
                        onVote={[Function]}
                        poll={
                                    Object {
                                                "_id": "1",
                                                "author": "123abc",
                                                "createdAt": 1000,
                                                "options": Array [
                                                  Object {
                                                    "_id": "101",
                                                    "value": "Yes",
                                                    "votes": 1,
                                                  },
                                                  Object {
                                                    "_id": "102",
                                                    "value": "No",
                                                    "votes": 0,
                                                  },
                                                ],
                                                "question": "Yes or No?",
                                                "voters": Array [
                                                  "123abc",
                                                ],
                                              }
                        }
            />
</div>,
          "className": "container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="level"
>
                <div
                                className="level-item"
                >
                                <h2
                                                className="title"
                                >
                                                Yes or No?
                                </h2>
                </div>
</div>,
              <div
                className="level"
>
                <div
                                className="level-item"
                >
                                <p
                                                className="subtitle"
                                >
                                                by 
                                </p>
                </div>
</div>,
              <div
                className="level"
>
                <div
                                className="level-item"
                />
</div>,
              <VotePoll
                onVote={[Function]}
                poll={
                                Object {
                                                "_id": "1",
                                                "author": "123abc",
                                                "createdAt": 1000,
                                                "options": Array [
                                                  Object {
                                                    "_id": "101",
                                                    "value": "Yes",
                                                    "votes": 1,
                                                  },
                                                  Object {
                                                    "_id": "102",
                                                    "value": "No",
                                                    "votes": 0,
                                                  },
                                                ],
                                                "question": "Yes or No?",
                                                "voters": Array [
                                                  "123abc",
                                                ],
                                              }
                }
/>,
            ],
            "className": "box",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="level-item"
>
                  <h2
                                    className="title"
                  >
                                    Yes or No?
                  </h2>
</div>,
                "className": "level",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <h2
                    className="title"
>
                    Yes or No?
</h2>,
                  "className": "level-item",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Yes or No?",
                    "className": "title",
                  },
                  "ref": null,
                  "rendered": "Yes or No?",
                  "type": "h2",
                },
                "type": "div",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="level-item"
>
                  <p
                                    className="subtitle"
                  >
                                    by 
                  </p>
</div>,
                "className": "level",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <p
                    className="subtitle"
>
                    by 
</p>,
                  "className": "level-item",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "by ",
                      undefined,
                    ],
                    "className": "subtitle",
                  },
                  "ref": null,
                  "rendered": Array [
                    "by ",
                    undefined,
                  ],
                  "type": "p",
                },
                "type": "div",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="level-item"
/>,
                "className": "level",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "level-item",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "onVote": [Function],
                "poll": Object {
                  "_id": "1",
                  "author": "123abc",
                  "createdAt": 1000,
                  "options": Array [
                    Object {
                      "_id": "101",
                      "value": "Yes",
                      "votes": 1,
                    },
                    Object {
                      "_id": "102",
                      "value": "No",
                      "votes": 0,
                    },
                  ],
                  "question": "Yes or No?",
                  "voters": Array [
                    "123abc",
                  ],
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
